cfme/infrastructure/resource_pool.py

```diff
@@ -53,72 +137,117 @@ def _get_context(self):
 
-        Args:
-            *ident: An InfoBlock title, followed by the Key name, e.g. "Relationships", "Images"
-        Returns: A string representing the contents of the InfoBlock's value.
+        :param *ident: An InfoBlock title, followed by the Key name, e.g. "Properties"
+        :returns: A string representing the contents of the InfoBlock's value.
         """
-        navigate_to(self, 'Details')
-        return details_page.infoblock.text(*ident)
+        view = navigate_to(self, 'Details')
+        table = None
+        if ident[0] == 'Properties':
+            table = view.properties
+        elif ident[0] == 'Relationships':
+            table = view.relationships
+        elif ident[0] == 'Smart Management':
+            table = view.smart_management
+        if table:
+            return table.get_text_of(ident[1])
+        return None
 
     @property
     def exists(self):
+        view = navigate_to(self, 'All')
         try:
-            navigate_to(self, 'All')
-            quad = Quadicon(self.name, self.quad_name)
-            if sel.is_displayed(quad):
-                return True
-        except sel.NoSuchElementException:
+            view.paginator.find_row_on_pages(view.entities.table, name=self.name)
+            return True
+        except NoSuchElementException:
             return False
 
 
 @navigator.register(ResourcePool, 'All')
 class All(CFMENavigateStep):
+    """A navigation step for the All page"""
+    VIEW = ResourcePoolAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def step(self):
+    def step(self, *args, **kwargs):
         self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Resource Pools')
 
     def resetter(self):
-        # Reset view and selection
-        tb.select("Grid View")
-        sel.check(paginator.check_all())
-        sel.uncheck(paginator.check_all())
+        """Reset view and selection"""
+        self.prerequisite_view.toolbar.select('Grid View')
+        self.prerequisite_view.paginator.check_all()
+        self.prerequisite_view.paginator.uncheck_all()
 
 
 @navigator.register(ResourcePool, 'Details')
 class Details(CFMENavigateStep):
+    """A navigation step for the Details page"""
+    VIEW = ResourcePoolDetailsView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        sel.click(Quadicon(self.obj.name, self.obj.quad_name))
+        """Navigate to the item"""
+        self.prerequisite_view.toolbar.view_selector.select('List View')
+        try:
+            row = self.prerequisite_view.paginator.find_row_on_pages(
+                self.prerequisite_view.entities.table,
+                name=self.obj.name
+            )
+        except NoSuchElementException:
+            raise ResourcePoolNotFound
+        row.click()
 
     def am_i_here(self):
-        return match_page(summary="{} (Summary)".format(self.obj.name))
+        return self.prerequisite_view.is_displayed
```

Comment:
**REQUIRED**
`am_i_here()` can be removed here, because `is_displayed()` is implemented for `ResourcePoolDetailsView`